lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -591,6 +712,36 @@ def is_ready(self) -> bool:
             self.on.patch_failed.emit(message=msg)
             return False
 
+    def get_status(self) -> Tuple[PatchState, str]:
```

Comment:
yeah I thought about that approach, but I felt giving the charms the control over which state they want to set on the charm instead of kind of imposing it on them. But, on second thought, they still have that control if they want to change the state we see as suitable. 
Will change the return type.

Reply:
> This practice is not yet well established in the ecosystem, and there aren't any conventions yet really, and some may argue that `get_status` methods should return `Sequence[StatusBase]` in complex libs, but for now I think we can start with 1:1 match with `add_status`, implying that a lib should have only one status at a time, even if a few things went wrong.

I feel like this is a good approach. The lib should present its combined conclusion on its status, having logged (or aggregated into a message) anything extra needed. The charm is already going to need to combine the lib status with ones from other libs or its own requirements; it's better to not move more of the complexity into the charm and instead leave that responsibility with the lib.

I think this would also evolve well in the future, if there are per-relation statuses as has been mooted (with the assumption that a lib generally 'owns' one/more relations), where the lib would set a combined status for each relation - it seems unlikely it would be able to break it down further than that.

Reply:
Yeah, I like @sed-i's suggestion here, makes it nice and easy for the caller to pass to `add_status()`, but doesn't actually call `add_status()`, so the caller could still disregard/tweak it if they wanted. I do think the simpler "single status" is better than "sequence of statuses".

Reply:
I'd love to comment, but I think I need to understand the resource patch concept first.